python/pyspark/sql/pandas/conversion.py

```diff
@@ -171,7 +171,18 @@ def toPandas(self) -> "PandasDataFrameLike":
                                 pdf[field.name] = _convert_map_items_to_dict(pdf[field.name])
                         return pdf
                     else:
-                        return pd.DataFrame.from_records([], columns=self.columns)
+                        corrected_panda_types = []
+                        for index, field in enumerate(self.schema):
+                            panda_type = PandasConversionMixin._to_corrected_pandas_type(
+                                field.dataType
+                            )
+                            if panda_type is None:
+                                corrected_panda_types.append((tmp_column_names[index], np.object0))
+                            else:
+                                corrected_panda_types.append((tmp_column_names[index], panda_type))
+                        pdf = pd.DataFrame(np.empty(0, dtype=corrected_panda_types), index=[])
+                        pdf.columns = self.columns
+                        return pdf
```

Comment:
nit: how about

```python
return pd.DataFrame(np.empty(0, dtype=corrected_panda_types), columns=self.columns)
```

??

Reply:
Agree , doing the change

Reply:
@itholic 
1) Providing columns at the time of creation of dataframe(as suggested) , changes all columns dtypes to object. Since self.columns can have duplicate column names , once the dataframe created then i have done the assignment
pdf.columns = self.columns . This way the dtypes are also remain intact(its similar way as done above in the code L#1654) 

2) We also need to provide index=[] , as otherwise its creating RangeIndex(start=0, stop=0, step=1). Empty DF , when created without arrow , in the below part of the code have index=[] , so in order to have exact empty DF (with and without arrow) , index is added

Reply:
Got it!

Then it looks good to me as is.